src/ray/raylet/worker_pool.cc

```diff
@@ -365,6 +375,13 @@ void WorkerPool::MonitorStartingWorkerProcess(const Process &proc,
         if (it != state.starting_worker_processes.end()) {
           RAY_LOG(INFO) << "Some workers of the worker process(" << proc.GetId()
                         << ") have not registered to raylet within timeout.";
+          auto task_id = it->second.task_id;
+          auto callback = it->second.callback;
+          if (!task_id.IsNil() && callback) {
+            callback(nullptr);
+            it->second.task_id = TaskID::Nil();
+            it->second.callback = PopWorkerCallback();
```

Comment:
What's the point of resetting these two fields? The entry will be erased anyway.

Reply:
If one worker process has multiple workers, it will not be erased.